components/match2/commons/match_group_input.lua

```diff
@@ -241,6 +243,39 @@ function MatchGroupInput.getInexactDate(suggestedDate)
 	return getContentLanguage():formatDate('c', inexactDateString)
 end
 
+--[[
+Parses the match group context. The match group context describes where a match
+group is relative to the tournament page.
+]]
+function MatchGroupInput.readContext(args)
+	return {
+		bracketIndex = tonumber(globalVars:get('match2bracketindex')) or 0,
+		groupRoundIndex = MatchGroupInput.readGroupRoundIndex(args),
+		matchSection = args.matchsection or globalVars:get('matchsection'),
+		sectionHeader = args.section or globalVars:get('bracket_header'),
+		tournamentParent = globalVars:get('tournament_parent'),
+	}
+end
+
+function MatchGroupInput.readGroupRoundIndex(args)
+	if args.round then
+		return tonumber(args.round)
+	end
+
+	local matchSection = args.matchsection or globalVars:get('matchsection')
+	-- Usually 'Round N' but can also be 'Day N', 'Week N', etc.
+	local roundIndex = matchSection and matchSection:match(' (%d+)$')
+	return roundIndex and tonumber(roundIndex)
+end
+
+--[[
+Saves a part of the match group context to page variables.
+]]
+function MatchGroupInput.persistContext(context)
```

Comment:
`persistContext` only saves a "part of" the context? Seems a bit weird?

Reply:
![image](https://user-images.githubusercontent.com/85348434/147249320-e99d9a58-2746-4e43-9458-4c013b40a8c2.png)

`sectionHeader` and `matchSection` are special because they can also be specified in the bracket args (`{{Matchlist|section=|matchsection=}}` `{{Bracket|section=|matchsection=}}`) When that happens they should be saved to page variables as if they were set from page variables.

Reply:
I renamed it to `persistContextChanges`